### PR TITLE
Feat: v3 PHP

### PR DESCRIPTION
**Pull request by Meldiron**
PHP implementation of V3: https://github.com/open-runtimes/open-runtimes/pull/136

### CODE REVIEW

File: runtimes/php-8.1/server.php [hunk 1]
```diff
@@ -3,96 +3,138 @@
...
+    $output['headers'] ??= [];
+
+    foreach ($output['headers'] as $header => $value) {
+        if(!(\str_starts_with(\strtolower($header), 'x-open-runtimes-'))) {
+            $res->header(\strtolower($header), $value);
+        }
+    }
+
+    if(!empty($customstd)) {
+        $context->log('Unsupported log noticed. Use context.log() or context.error() for logging.');
```
Comment (Meldiron) [inline runtimes/php-8.1/server.php:130]: Improve message

File: runtimes/php-8.1/server.php [hunk 2]
```diff
@@ -3,96 +3,138 @@
...
+        $customstd = ob_get_clean();
     } catch (\Throwable $e) {
-        $res->status(500);
-        return $res->end(json_encode(['stderr' => $e->getMessage()."\r\n".$e->getTraceAsString()], JSON_UNESCAPED_UNICODE|JSON_UNESCAPED_SLASHES));
+        $context->error($e->getMessage()."\n".$e->getTraceAsString());
+        $output = $context->res->send('', 500, []);
+    }
+
+    if($output == null) {
+        $context->error('Return statement missing. return context.res.empty() if no response is expected.');
```
Comment (Meldiron) [inline runtimes/php-8.1/server.php:115]: Improve message

**Review by lohanidamodar — COMMENTED** [review]
Apart from Node and python changes (not sure why they are in this PR), this looks good.

**Comment by Meldiron** [comment]
Node and python changes removed from PR.

File: runtimes/php-8.1/server.php [hunk 3]
```diff
@@ -3,96 +3,191 @@
 
 const USER_CODE_PATH = '/usr/code-start';
 
-function join_paths() {
-    $paths = array();
+class Response {
```
Comment (abnegate) [inline runtimes/php-8.1/server.php:6]: We should prefix the class names with `Runtime` for consistency across runtimes

File: runtimes/php-8.1/server.php [hunk 4]
```diff
@@ -3,96 +3,191 @@
...
+            $this->logs[] = \strval($message);
+        }
     }
 
-    function json($json, $status = 200) {
-        $this->status = $status;
-        $this->response = $json;
-        $this->headers['Content-Type'] = 'application/json';
+    function error(mixed $message) {
+        if(\is_array($message)) {
```
Comment (abnegate) [inline runtimes/php-8.1/server.php:65]: Do we want to check for `is_object` here too?

Comment (Meldiron) [reply runtimes/php-8.1/server.php:65]: Agree. added.

File: runtimes/php-8.1/server.php (hunk 3 above)
Comment (Meldiron) [reply runtimes/php-8.1/server.php:6]: Updated.

File: runtimes/php-8.1/README.md [hunk 5]
```diff
@@ -72,26 +72,8 @@ You can also make changes to the example code and apply the changes with the `do
...
-    $res->send('Hello Open Runtimes 👋');
-};
-```
-
-- The `res` parameter has two methods:
-
-    - `send()`: Send a string response to the client.
-    - `json()`: Send a JSON response to the client.
-
-You can respond with `json()` by providing object:
```
Comment (christyjacob4) [inline runtimes/php-8.1/README.md:85]: Why did we remove this ? Shouldn't we still keep the documentation for the `context` object ?

Comment (Meldiron) [reply runtimes/php-8.1/README.md:85]: It's consistent, so I removed it from runtime readme. I want to add it to main README.

**Review by christyjacob4 — APPROVED** [review]

File: runtimes/php-8.1/README.md (hunk 5 above)
Comment (christyjacob4) [reply runtimes/php-8.1/README.md:85]: okay 👍

**Pull request merged**